services/upgrader/upgrader_linux.go

```diff
@@ -256,6 +256,15 @@ func download(url string, limit int64) (string, error) {
 	}
 	defer resp.Body.Close()
 
+	if resp.ContentLength > limit {
+		mlog.Warn("ContentLength exceeds the download limit", mlog.String("url", url), mlog.Int64("limit", limit), mlog.Int64("contentLength", resp.ContentLength))
```

Comment:
The ContentLength is reported by the server. It _might_ not be accurate, so we just warn here.